### PR TITLE
MessageAdd, Multiply, Divide, Pow and Subtract

### PR DESCRIPTION
**Pull request by Kyew**
Now the variable part of the computation is saved so that the 'last' value can be recomputed when the 'constant' parts changes, as it does in pd.

### CODE REVIEW

**Comment by mhroth** [comment]
Does Pd really do this? Pd is so stupid sometimes.

**Comment by Kyew** [comment]
Yes it does ^^

2014-11-05 0:02 GMT+01:00 Martin Roth notifications@github.com:

> Does Pd really do this? Pd is so stupid sometimes.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mhroth/ZenGarden/pull/54#issuecomment-61731402.

## 

SAUVAGEOT Paul-Arthur.
Développeur C \ C++ \ C#.

File: src/MessageDivide.cpp [hunk 1]
```diff
@@ -46,7 +47,8 @@ void MessageDivide::processMessage(int inletIndex, PdMessage *message) {
     case 0: {
       switch (message->getType(0)) {
         case FLOAT: {
-          last = constant == 0.0f ? 0.0f : message->getFloat(0) / constant;
+          variable = message->getFloat(0);
+          last = constant == 0.0f ? 0.0f : variable / constant;
```
Comment (mhroth) [inline src/MessageDivide.cpp]: For tertiary operator, I would prefer a syntax with parenthesis for clarity `last = (constant == 0.0f) ? 0.0f : variable / constant;`

File: src/MessageDivide.cpp [hunk 2]
```diff
@@ -62,6 +64,7 @@ void MessageDivide::processMessage(int inletIndex, PdMessage *message) {
     case 1: {
       if (message->isFloat(0)) {
         constant = message->getFloat(0);
+        last = constant == 0.0f ? 0.0f : variable / constant;
```
Comment (mhroth) [inline src/MessageDivide.cpp]: See above.

**Pull request closed without merging**